### PR TITLE
Rename bubblegum accounts to make more obvious

### PR DESCRIPTION
**Pull request by danenbm**
**Note this PR is stacked onto the changes for https://github.com/metaplex-foundation/metaplex-program-library/pull/728**

### Notes
- Reorder account validation structs into somewhat consistent
order.
- Change any instances of `candy_wrapper` to `log_wrapper`.
- Change any instances of `merkle_roll` to `merkle_tree`.
- Rename `authority` to `tree_authority`.
- Rename `owner` and `delegate` to `leaf_owner` and either
`leaf_delegate` or `tree_delegate` depending on context.

### CODE REVIEW

File: bubblegum/program/src/state/mod.rs [hunk 1]
```diff
@@ -16,8 +16,8 @@ pub const COLLECTION_CPI_PREFIX: &str = "collection_cpi";
 #[account]
 #[derive(Copy)]
 pub struct TreeConfig {
-    pub creator: Pubkey,
-    pub delegate: Pubkey,
+    pub tree_creator: Pubkey,
```
Comment (danenbm) [inline bubblegum/program/src/state/mod.rs:19]: I know this looks redundant but it helps with the anchor constraints.

**Review by stranzhay — APPROVED** [review]
i like the more descriptive naming. lgtm.

File: bubblegum/program/src/lib.rs [hunk 2]
```diff
@@ -785,9 +786,9 @@ fn process_collection_verification<'info>(
         &merkle_tree.key(),
         *ctx.bumps.get("authority").unwrap(),
```
Comment (alexandruag) [inline bubblegum/program/src/lib.rs:787]: Does this now need to be "tree_authority"?

Comment (danenbm) [reply bubblegum/program/src/lib.rs:787]: yes good catch I knew I needed to find and replace all of those I even have a note here to do that in my text editor, but forgot to do that.

Comment (danenbm) [reply bubblegum/program/src/lib.rs]: Fixed

**Review by alexandruag — APPROVED** [review]

**Pull request merged**